lightning/src/ln/wire.rs

```diff
@@ -13,7 +13,7 @@
 //! The [`Message`] enum returned by [`read()`] wraps the decoded message or the message type (if
 //! unknown) to use with pattern matching.
 //!
-//! Messages implementing the [`Encode`] trait define a message type and can be sent over the wire
+//! Messages implementing the [`Type`] trait define a message type and can be sent over the wire
```

Comment:
These docs should be updated to only describe the public interface, currently `cargo doc` says:

```
warning: public documentation for `wire` links to private item `read`
  --> lightning/src/ln/wire.rs:12:68
   |
12 | //! Messages known by this module can be read from the wire using [`read()`].
   |                                                                    ^^^^^^^^ this item is private
   |
   = note: `#[warn(private_intra_doc_links)]` on by default
   = note: this link will resolve properly if you pass `--document-private-items`

warning: public documentation for `wire` links to private item `Message`
  --> lightning/src/ln/wire.rs:13:10
   |
13 | //! The [`Message`] enum returned by [`read()`] wraps the decoded message or the message type (if
   |          ^^^^^^^^^ this item is private
   |
   = note: this link will resolve properly if you pass `--document-private-items`

warning: public documentation for `wire` links to private item `read`
  --> lightning/src/ln/wire.rs:13:39
   |
13 | //! The [`Message`] enum returned by [`read()`] wraps the decoded message or the message type (if
   |                                       ^^^^^^^^ this item is private
   |
   = note: this link will resolve properly if you pass `--document-private-items`

warning: 3 warnings emitted
```

Reply:
Updated the docs to remove private item mentions.